bpfassets/perf_event/perf_event.c

```diff
@@ -206,7 +207,7 @@ int kprobe__finish_task_switch(struct pt_regs *ctx, struct task_struct *prev)
 
     // store process metrics
     struct process_metrics_t *process_metrics;
-    process_metrics = processes.lookup(&cur_pid);
+    process_metrics = processes.lookup(&prev_pid);
```

Comment:
the `new_process.pid = cur_pid;` should be:
`new_process.pid = prev_pid;`

after this change we can merge this PR

Reply:
updated. Now the `new_process.pid` is only created for cur_pid.